cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -3937,7 +3937,7 @@ TEST(Substrait, ProjectWithMultiFieldExpressions) {
             }]
           }
         },
-        "names": ["A", "B", "C", "D"]
```

Comment:
Why this change?

Reply:
This seems wrong. The plan has 4 output columns:
direct reference 0, 1, 2 and a new column with scalar Function

Reply:
Without this change, [this check](https://github.com/apache/arrow/pull/35364/files#diff-11e93118405fd9709ff4ffb62aa7fbbc6ff8ca5bee7aa34f8a0ae951360d7a6eR158-R161) raises. IIUC, the plan has 3 outputs because its root is a projection with 3 emit indices.

Reply:
I see

Reply:
These names here (A, B, C) doesn't match the “output_schema" below (A, A1, A+1). Is this expected?

Reply:
(Not that I expect your code change has anything to do with this, but looks quite confusing to me so I asked)

Reply:
Right, not related to my changes herer. The invocations look like `CheckRoundTripResult` -> `AssertTablesEqualIgnoringOrder` -> `AssertTablesEqual`, and the latter (surprisingly, I guess) ignores column names for some reason.